diff --git a/src/node_binding.cc b/src/node_binding.cc
index 8665b28275..ef93acc75a 100644
--- a/src/node_binding.cc
+++ b/src/node_binding.cc
@@ -48,6 +48,7 @@
   V(fs_event_wrap)                                                             \
   V(heap_utils)                                                                \
   V(http2)                                                                     \
+  V(python)                                                                    \
   V(http_parser)                                                               \
   V(inspector)                                                                 \
   V(js_stream)                                                                 \
