diff --git a/src/node_binding.cc b/src/node_binding.cc
index 6c7ab4b21e..0cf8a5f9a0 100644
--- a/src/node_binding.cc
+++ b/src/node_binding.cc
@@ -52,6 +52,7 @@
   V(fs_event_wrap)                                                             \
   V(heap_utils)                                                                \
   V(http2)                                                                     \
+  V(python)                                                                    \
   V(http_parser)                                                               \
   V(inspector)                                                                 \
   V(js_stream)                                                                 \
