diff --git a/node.gyp b/node.gyp
index 63c8c50987..b884a0ff2e 100644
--- a/node.gyp
+++ b/node.gyp
@@ -47,6 +47,7 @@
       'lib/console.js',
       'lib/constants.js',
       'lib/crypto.js',
+      'lib/python.js',
       'lib/cluster.js',
       'lib/diagnostics_channel.js',
       'lib/dgram.js',
@@ -679,6 +680,35 @@
         'src/udp_wrap.cc',
         'src/util.cc',
         'src/uv.cc',
+            'src/python_node.cc',
+            'src/python_node.h',
+            'src/WeakValueMap.cc',
+            'src/WeakValueMap.h',
+            'src/node_ref.h',
+            'src/python_bind.cc',
+            'src/python_bind.h',
+            'src/CustomModules.cc',
+            'src/CustomModules.h',
+            'src/JsVars.cc',
+            'src/JsVars.h',
+            'src/PyVars.cc',
+            'src/PyVars.h',
+            'src/wait_promise.cc',
+            'src/wait_promise.h',
+            'src/uibox.cc',
+            'src/uibox.h',
+            'src/pyconverters.cc',
+            'src/pyconverters.h',
+            'src/bg_task.cc',
+            'src/bg_task.h',
+            'src/call_later.cc',
+            'src/call_later.h',
+            'src/JsObject.cc',
+            'src/JsObject.h',
+            'src/JsCallObject.cc',
+            'src/JsCallObject.h',
+            'src/PlynthUtils.cc',
+            'src/PlynthUtils.h',
         # headers to make for a more pleasant IDE experience
         'src/aliased_buffer.h',
         'src/aliased_struct.h',
diff --git a/src/node_binding.cc b/src/node_binding.cc
index 6c7ab4b21e..0cf8a5f9a0 100644
--- a/src/node_binding.cc
+++ b/src/node_binding.cc
@@ -52,6 +52,7 @@
   V(fs_event_wrap)                                                             \
   V(heap_utils)                                                                \
   V(http2)                                                                     \
+  V(python)                                                                    \
   V(http_parser)                                                               \
   V(inspector)                                                                 \
   V(js_stream)                                                                 \
